commands/registry_login.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) OpenFaaS Author(s) 2020. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var registryLoginCommand = &cobra.Command{
+	Use:          "registry-login",
+	Short:        "Generate and save the registry authentication file",
+	SilenceUsage: true,
+	RunE:         generateRegistryAuthFile,
+}
+
+func init() {
+	registryLoginCommand.Flags().String("server", "https://index.docker.io/v1/", "The server URL, it is defaulted to the docker registry")
+	registryLoginCommand.Flags().StringP("username", "u", "", "The Registry Username")
+	registryLoginCommand.Flags().String("password", "", "The registry password")
+	registryLoginCommand.Flags().BoolP("password-stdin", "s", false, "Reads the docker password from stdin, either pipe to the command or remember to press ctrl+d when reading interactively")
+
+	registryLoginCommand.Flags().Bool("ecr", false, "If we are using ECR we need a different set of flags, so if this is set, we need to set --username and --password")
+	registryLoginCommand.Flags().String("account-id", "", "Your AWS Account id")
+	registryLoginCommand.Flags().String("region", "", "Your AWS region")
+
+	faasCmd.AddCommand(registryLoginCommand)
+}
+
+func generateRegistryAuthFile(command *cobra.Command, _ []string) error {
+	ecrEnabled, _ := command.Flags().GetBool("ecr")
```

Comment:
We need error handling on these flags

Reply:
Thanks for mentioning, have handled error check in `PreRunE`